### DISCUSSION TITLE
SVG support

### DISCUSSION THREAD

**Discussion by GarethLangleyQuitGenius** [JonasKruckenberg/imagetools#387] [Q&A]
Does this library support converting svgs to webp? I cannot see anything in the docs that specifies what file types can be converted.

**Comment by benmccann** [comment]
I would think an svg would normally be smaller than a webp, so there wouldn't be much point in that?

**Reply by GarethLangleyQuitGenius** [reply]
I don't believe that this is always the case, for instance I have a svg file that is 524 KB and when I convert to WebP manually it reduces the file size down to 29 KB.

**Reply by JonasKruckenberg** [reply]
SVGs serve much different use cases than raster-based images and you generally shouldn't convert them into rasterized formats. 

But to answer your original question: Yes. Loading an svg as an input should be possible (I have never tested this myself but sharp accepts svgs)

But again **this doesn't make much sense** if you want smaller svgs use a tool like [svgo](https://github.com/svg/svgo)

**Reply by TGlide** [reply]
It's still useful in certain cases besides lower file formats, e.g. you have download buttons for assets inside your website in multiple formats

**Comment by la-magra** [comment]
Have you tried to set the following imagetools config?

```js
// vite.config.js
// ...
return defineConfig({
// ...
  plugins: [
    imagetools({
      include: [
      '**/*.{heic,heif,avif,jpeg,jpg,png,tiff,webp,gif,svg}', // svg added
      '**/*.{heic,heif,avif,jpeg,jpg,png,tiff,webp,gif,svg}?*', // svg added
      ],
    }),
  ],
// ...
```

```shell
vite v4.1.4 building for production...
✓ 304 modules transformed.
../dist/index.html                                 0.47 kB
../dist/assets/logo-7d9f048f.webp                  8.83 kB  # this was a complex 152KiB svg that was already optimized by svgo
```

**Reply by la-magra** [reply]
Just as an update:

`include` now requires a string, use this to process svgs too:
```javascript
include: /^[^?]+\.(heic|heif|avif|jpeg|jpg|png|tiff|webp|gif|svg)(\?.*)?$/
```
